contracts/SpokePool.sol

```diff
@@ -1094,8 +1095,92 @@ abstract contract SpokePool is
         uint32 rootBundleId,
         SpokePoolInterface.RelayerRefundLeaf memory relayerRefundLeaf,
         bytes32[] memory proof
-    ) public virtual override nonReentrant {
-        _executeRelayerRefundLeaf(rootBundleId, relayerRefundLeaf, proof);
+    ) public override nonReentrant {
+        _preExecuteLeafHook(relayerRefundLeaf.l2TokenAddress);
+
+        _validateRelayerRefundLeaf(
+            relayerRefundLeaf.chainId,
+            relayerRefundLeaf.refundAddresses,
+            relayerRefundLeaf.refundAmounts
+        );
+
+        RootBundle storage rootBundle = rootBundles[rootBundleId];
+
+        // Check that inclusionProof proves that relayerRefundLeaf is contained within the relayer refund root.
+        // Note: This should revert if the relayerRefundRoot is uninitialized.
+        require(MerkleLib.verifyRelayerRefund(rootBundle.relayerRefundRoot, relayerRefundLeaf, proof), "Bad Proof");
+
+        _setClaimedLeaf(rootBundleId, relayerRefundLeaf.leafId);
+
+        _distributeRelayerRefunds(
+            relayerRefundLeaf.amountToReturn,
+            relayerRefundLeaf.refundAmounts,
+            relayerRefundLeaf.leafId,
+            relayerRefundLeaf.l2TokenAddress,
+            relayerRefundLeaf.refundAddresses
+        );
+
+        emit ExecutedRelayerRefundRoot(
+            relayerRefundLeaf.amountToReturn,
+            relayerRefundLeaf.chainId,
+            relayerRefundLeaf.refundAmounts,
+            rootBundleId,
+            relayerRefundLeaf.leafId,
+            relayerRefundLeaf.l2TokenAddress,
+            relayerRefundLeaf.refundAddresses,
+            msg.sender
+        );
+    }
+
+    /**
+     * @notice Executes a relayer refund leaf stored as part of a root bundle. Will send the relayer the amount they
+     * sent to the recipient plus a relayer fee.
+     * @param rootBundleId Unique ID of root bundle containing relayer refund root that this leaf is contained in.
+     * @param relayerRefundLeaf Contains all data necessary to reconstruct leaf contained in root bundle and to
+     * refund relayer. This data structure is explained in detail in the SpokePoolInterface.
+     * @param proof Inclusion proof for this leaf in relayer refund root in root bundle.
+     */
+    function executeRelayerRefundLeafUSS(
```

Comment:
I did try refactoring as much as possible, how would you condense more? Note that I wanted to preserve the order of verifying the MerkleLeaf is included and unclaimed before distributing funds. This means that we need to make a call to `MerkleLib.verifyRelayerRefundLeaf{USS}()` in the middle of two internal function calls at the least

Reply:
I was just thinking about factoring out the code before and after the `require()` statement, but it might be a case of diminishing returns since the implementation is pretty simple.